.travis.yml

```diff
@@ -1,26 +1,92 @@
+#     the ros wiki). If the file does not exists then nothing happens.
+#
+# See the README.md for more information.
+#
+# Author: Felix Duvallet <felixd@gmail.com>
+
+# NOTE: The build lifecycle on Travis.ci is something like this:
+#    before_install
+#    install
+#    before_script
+#    script
+#    after_success or after_failure
+#    after_script
+#    OPTIONAL before_deploy
+#    OPTIONAL deploy
+#    OPTIONAL after_deploy
 
+################################################################################
+
+# Use ubuntu trusty (14.04) with sudo privileges.
+dist: trusty
+sudo: required
 language:
-    c++
+  - python
+  - c++
+cache:
+  - apt
+
+# By default travis runs all python code in a virtualenv that does not contain
+# the packages installed with apt-get. As this is essential (pip doesn't contain
+# all the necessary ROS python packages), tell travis-ci to use the system-wide
+# python packages.
+virtualenv:
+  system_site_packages: true
+
+# Configuration variables. All variables are global now, but this can be used to
+# trigger a build matrix for different ROS distributions if desired.
+env:
+  global:
+    - ROS_DISTRO=indigo
+    - ROS_CI_DESKTOP="`lsb_release -cs`"  # e.g. [precise|trusty|...]
+    - ROS_PARALLEL_JOBS='-j8 -l6'
+
+################################################################################
 
-compiler:
-    - gcc
+# Install system dependencies, namely a very barebones ROS setup.
+before_install:
+  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
+  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
+  - sudo apt-get update -qq
+  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin ros-$ROS_DISTRO-desktop
+  - sudo apt-get install -y ros-$ROS_DISTRO-pcl-ros ros-$ROS_DISTRO-image-transport ros-$ROS_DISTRO-image-transport-plugins ros-$ROS_DISTRO-libg2o
+    # Prepare rosdep to install dependencies.
+  - sudo rosdep init
+  - rosdep update
+  - source /opt/ros/$ROS_DISTRO/setup.bash
+  - sudo apt-get install python-rosinstall
 
+# Install all dependencies, using wstool and rosdep.
+# wstool looks for a ROSINSTALL_FILE defined in the environment variables.
 before_script:
-    - sudo bash scripts/ros_install.bash
-    - sudo bash scripts/ci_install.bash
+- bash scripts/mcptam_install.bash
 
+# Compile and test. If the CATKIN_OPTIONS file exists, use it as an argument to
+# catkin_make.
 script:
-    - echo "done!"
-
-branches:
-    only:
-        - issue_6
-
-notifications:
-    recipients:
-        - chutsu@gmail.com
-    email:
-        on_success: change
-        on_failure: always
+  - source $HOME/catkin_ws/devel/setup.bash
+  - cd ~/catkin_ws
+  - catkin_make 
+  # Testing: Use both run_tests (to see the output) and test (to error out).
+  - catkin_make run_tests  # This always returns 0, but looks pretty.
```

Comment:
changed the testing order, we now run 

catkin_make test to build and show the output of the tests.  The output is displayed nicely on the console so you can see quickly where the test failed.  Then we run

catkin_test_results, which outputs the results, and exits non-zero if there's a failure